src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -161,7 +161,7 @@ class PrisonerIepLevelReviewService(
         val iepHistory = getPrisonerIepLevelHistory(prisonerInfo.bookingId, useNomisData = true, withDetails = true, useClientCredentials = true).iepDetails
         val iepLevelBeforeTransfer = iepHistory.sortedBy(IepDetail::iepTime).lastOrNull { it.agencyId != prisonerInfo.agencyId }
           ?: throw NoDataFoundException(prisonerInfo.bookingId)
-        iepLevelsForPrison.find { it.iepLevel == iepLevelBeforeTransfer.iepLevel }
+        iepLevelsForPrison.find { it.iepDescription == iepLevelBeforeTransfer.iepLevel }
```

Comment:
When updating the test `PrisonerIepLevelReviewServiceTest.process TRANSFERRED when offender stays on same incentive level as previous prison` I realised we were setting `IepDetail.iepLevel` to the abbreviated format of an IEP level. For example `STD` rather than `Standard`

We should be comparing either
`IepLevel.iepDescription` with `IepDetail.iepLevel`
or
`IepLevel.iepLevel` with `IepDetail.iepCode`

Reply:
This makes sense.

However [the `IepLevel` class](https://github.com/ministryofjustice/hmpps-incentives-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiDto.kt#L42-L51) is a bit evil here as its fields are named differently and caused this understandable confusion:

```kotlin
data class IepLevel(
  @Schema(description = "IEP Code for an IEP level", example = "STD")
  val iepLevel: String,
  @Schema(description = "Description of the IEP Level", example = "Standard")
  val iepDescription: String,
  // [...]
```

In other places `iepLevel` is the description (e.g. `"Standard"`) and `iepCode` is the code (e.g. `"STD"`)...but not in that class :(

I don't think that class is exposed in any of our API responses. I can't quite tell if it's used to deserialise a response from Prison API.

I think it would be ideal to update this class to make it consistent to the others and make the `iepLevel` the description and `iepCode` the code (if it doesn't break compatibility of our API/Prison API responses)

Reply:
OK, I was playing with this locally and it seems [there is at least one place in the API we use this `IepLevel` class to construct the response (`GET /iep/levels/{prisonId}`)](https://github.com/ministryofjustice/hmpps-incentives-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/resource/IepLevelResource.kt#L39-L76), so no, we cannot rename these fields without potentially breaking compatibility with some of the consumers 🤷

Reply:
for what it's worth, DPS uses `iepLevel` fields (with all their different meanings) from several endpoints so we're kinda stuck with this mixup now